containers/Java/.devcontainer/devcontainer.json

```diff
@@ -0,0 +1,54 @@
+// For format details, see https://aka.ms/devcontainer.json. For config options, see the README at:
+// https://github.com/microsoft/vscode-dev-containers/tree/v0.194.0/containers/java
+{
+    "name": "Java",
+    "build": {
+        "dockerfile": "Dockerfile",
+        "args": {
+            // Update the VARIANT arg to pick a Java version: 11, 16
+            "VARIANT": "11",
+            // Options
+            "INSTALL_MAVEN": "true",
+            "INSTALL_GRADLE": "false",
+            "NODE_VERSION": "none"
+        }
+    },
+
+    "runArgs": [
+        "--network",
+        "host"
+    ],
+
+    "mounts": [
+        // Keep command history 
+        "source=ostf-bashhistory,target=/commandhistory,type=volume",
+        // Use hosts docker socket
+        "source=/var/run/docker.sock,target=/var/run/docker.sock,type=bind"
+    ],
+
+    // Set *default* container specific settings.json values on container create.
+    "settings": { 
+        "java.home": "/docker-java-home",
+        "terminal.integrated.profiles.linux": {
+            "bash": {
+                "path": "/bin/bash"
+            }
+        }
+    },
+
+    // Add the IDs of extensions you want installed when the container is created.
+    "extensions": [
+        "vscjava.vscode-java-pack",
+        "ms-vscode.azure-account",
+        "ms-azuretools.vscode-docker",
+        "vsciot-vscode.azure-iot-toolkit",
+        "vsciot-vscode.azure-iot-tools",
+        "vsciot-vscode.vscode-iot-device-cube"
```

Comment:
we should only need java, docker nad iottools extensions. The rest are pulled in as dependencies